tests/unit-tests/gconstruct/test_construct_graph.py

```diff
@@ -327,6 +332,75 @@ def test_feat_ops():
     assert_almost_equal(proc_res6[:,0], data_col0)
     assert_almost_equal(proc_res6[:,1], data_col1)
 
+def check_feat_ops_categorical():
+    feat_op7 = [
+        {
+            "feature_col": "test1",
+            "feature_name": "test7",
+            "transform": {"name": 'to_categorical'},
+        },
+    ]
+    res7 = parse_feat_ops(feat_op7)
+    data0 = {
+        "test1": [str(i) for i in np.random.randint(0, 10, size=10)] + [str(i) for i in range(10)],
+    }
+    data1 = {
+        "test1": [str(i) for i in np.random.randint(0, 10, size=10)],
+    }
+    preproc_res0 = preprocess_features(data0, res7)
+    preproc_res1 = preprocess_features(data1, res7)
+    assert "test7" in preproc_res0
+    assert "test7" in preproc_res1
+    return_dict = {
+        0: preproc_res0,
+        1: preproc_res1
+    }
+    update_two_phase_feat_ops(return_dict, res7)
+    proc_res3 = process_features(data0, res7)
+    assert "test7" in proc_res3
+    for one_hot, str_i in zip(proc_res3["test7"], data0["test1"]):
+        assert one_hot[int(str_i)] == 1
+    assert 'mapping' in feat_op7[0]["transform"]
+
```

Comment:
Can we add a test as passing feat_op7[0]["transform"]['mapping'] to categorical transform and do the data transform again on data0 and data1 and campare the new `one_hot` with the one in L361?